Profiler/Core/ProfilerToken.cs

```diff
@@ -1,16 +1,39 @@
-using System.Diagnostics;
+using System;
+using VRage.ModAPI;
 
 namespace Profiler.Core
 {
+    /// <summary>
+    /// Mark the beginning of a method profiling and be consumed in the end of profiling.
+    /// </summary>
     internal readonly struct ProfilerToken
     {
-        public readonly ProfilerEntry Entry;
-        public readonly long Start;
+        /// <summary>
+        /// Game entity responsible for the profiled method.
+        /// </summary>
+        public readonly IMyEntity GameEntity;
 
-        public ProfilerToken(ProfilerEntry entry)
+        /// <summary>
+        /// Entrypoint of the profiled method.
+        /// </summary>
+        public readonly string Entrypoint;
+
+        /// <summary>
+        /// Timestamp of when this profiling started.
+        /// </summary>
+        public readonly DateTime StartTimestamp;
```

Comment:
You should continue using the raw time in ticks instead of the relatively heavy DateTime object.